plugins/modules/proxmox_vm_info.py

```diff
@@ -0,0 +1,216 @@
+          "vmid": 100
+        },
+        {
+          "cpu": 0,
+          "cpus": 1,
+          "disk": 0,
+          "diskread": 0,
+          "diskwrite": 0,
+          "maxdisk": 0,
+          "maxmem": 536870912,
+          "mem": 0,
+          "name": "test1",
+          "netin": 0,
+          "netout": 0,
+          "status": "stopped",
+          "type": "qemu",
+          "uptime": 0,
+          "vmid": 101
+        }
+      ]
+"""
+
+from ansible.module_utils.basic import AnsibleModule
+from ansible_collections.community.general.plugins.module_utils.proxmox import (
+    proxmox_auth_argument_spec,
+    ProxmoxAnsible,
+)
+
+
+class ProxmoxVmInfoAnsible(ProxmoxAnsible):
+    def get_qemu_vms(self, node, vmid=None):
+        try:
+            vms = self.proxmox_api.nodes(node).qemu().get()
+            for vm in vms:
+                vm["vmid"] = int(vm["vmid"])
+                vm["type"] = "qemu"
+            if vmid is None:
+                return vms
+            return [vm for vm in vms if vm["vmid"] == vmid]
+        except Exception as e:
+            self.module.fail_json(msg="Failed to retrieve QEMU VMs information: %s" % e)
+
+    def get_lxc_vms(self, node, vmid=None):
+        try:
+            vms = self.proxmox_api.nodes(node).lxc().get()
+            for vm in vms:
+                vm["vmid"] = int(vm["vmid"])
+            if vmid is None:
+                return vms
+            return [vm for vm in vms if vm["vmid"] == vmid]
+        except Exception as e:
+            self.module.fail_json(msg="Failed to retrieve LXC VMs information: %s" % e)
+
+
+def main():
+    module_args = proxmox_auth_argument_spec()
+    vm_info_args = dict(
+        node=dict(type="str", required=True),
+        type=dict(
+            type="str", choices=["lxc", "qemu", "all"], default="all", required=False
+        ),
+        vmid=dict(type="int", required=False),
+        name=dict(type="str", required=False),
+    )
+    module_args.update(vm_info_args)
+
+    module = AnsibleModule(
+        argument_spec=module_args,
+        required_together=[("api_token_id", "api_token_secret")],
+        required_one_of=[("api_password", "api_token_id")],
+        supports_check_mode=True,
+    )
+
+    proxmox = ProxmoxVmInfoAnsible(module)
+    node = module.params["node"]
+    type = module.params["type"]
+    vmid = module.params["vmid"]
+    name = module.params["name"]
+
+    result = dict(changed=False)
+
+    if proxmox.get_node(node) is None:
+        module.fail_json(msg="Node %s doesn't exist in PVE cluster" % node)
+
+    if not vmid and name:
+        vmid = int(proxmox.get_vmid(name, ignore_missing=False))
+
+    vms = None
+    if type == "lxc":
+        vms = proxmox.get_lxc_vms(node, vmid=vmid)
+    elif type == "qemu":
+        vms = proxmox.get_qemu_vms(node, vmid=vmid)
+    else:
+        vms = proxmox.get_qemu_vms(node, vmid=vmid) + proxmox.get_lxc_vms(
+            node, vmid=vmid
+        )
+
+    if vms:
```

Comment:
If the user asked for a list of all VMs, and the list happens to be empty, the module right now fails with "VM with vmid None doesn't exist on node xxx".
```suggestion
    if vms or vmid is None:
```